### PR TITLE
5-24-19

### PR DESCRIPTION
**Pull request by DCamaron1**
started and finished into to hash maps

### CODE REVIEW

File: src/_06_Intro_To_Hash_Maps/_01_IntroToHashMaps.java [hunk 1]
```diff
@@ -4,17 +4,12 @@
...
-		//   current iteration of the loop (i). The value entry will be the square root of i.
-
-		//3. Iterate through all the entries in your HashMap displaying the keys with their respective
-		//   square roots (values). Use the following format.
-
-		//   The square Root of 0 is 0.0
-		//   The square Root of 1 is 1.0
-		//   The square Root of 2 is 1.4142135623730951
-		//   The square Root of 3 is 1.7320508075688772
+		HashMap<Integer, Double> roots = new HashMap<Integer, Double>();
```
Comment (IsisJ) [inline src/_06_Intro_To_Hash_Maps/_01_IntroToHashMaps.java]: I don't think that you need to put anything in the diamond operator when you instantiate/initialize a HashMap

File: src/_06_Intro_To_Hash_Maps/_02_LogSearch.java [hunk 2]
```diff
@@ -1,33 +1,80 @@
...
-	 *
-	 * */
-	
+import javax.swing.JButton;
+import javax.swing.JFrame;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+
+public class _02_LogSearch implements ActionListener {
+	HashMap<Integer, String> map = new HashMap<Integer, String>();
```
Comment (IsisJ) [inline src/_06_Intro_To_Hash_Maps/_02_LogSearch.java]: I don't think that you need to put anything in the diamond operator when you instantiate/initialize a HashMap 🍶

File: src/_06_Intro_To_Hash_Maps/_02_LogSearch.java [hunk 3]
```diff
@@ -1,33 +1,80 @@
...
+	private JFrame frame = new JFrame();
+	private JPanel panel = new JPanel();
+	private JButton addButton = new JButton();
+	private JButton searchButton = new JButton();
+	private JButton viewButton = new JButton();
+	private JButton removeButton = new JButton();
+
```
Comment (IsisJ) [inline src/_06_Intro_To_Hash_Maps/_02_LogSearch.java:20]: I like how descriptive your names are 👍

File: src/_06_Intro_To_Hash_Maps/_02_LogSearch.java [hunk 4]
```diff
@@ -1,33 +1,80 @@
...
+	public void actionPerformed(ActionEvent e) {
+		JButton buttonPressed = (JButton) e.getSource();
+		if (buttonPressed == addButton) {
+			String userID = JOptionPane.showInputDialog(null, "Enter an ID number (3 digits)");
+			String userName = JOptionPane.showInputDialog(null, "Enter a name");
+			Integer userIDInteger = Integer.parseInt(userID);
+			map.put(userIDInteger, userName);
+		}
+		if (buttonPressed == searchButton) {
+			String userID = JOptionPane.showInputDialog(null, "Enter an ID number (3 digits)");
```
Comment (IsisJ) [inline src/_06_Intro_To_Hash_Maps/_02_LogSearch.java:64]: Just a suggestion....but, you could reword the string in your input dialogue to say " Enter a 3 digit ID number" 🐑

File: src/_06_Intro_To_Hash_Maps/_02_LogSearch.java [hunk 5]
```diff
@@ -1,33 +1,80 @@
...
+				JOptionPane.showMessageDialog(null, "This username does not exist");
+			}
+
+		}
+		if (buttonPressed == viewButton) {
+			for (Integer i : map.keySet()) {
+				System.out.println("ID: " + i + " Name: " + map.get(i));
+			}
+		}
+		if (buttonPressed == removeButton) {
```
Comment (IsisJ) [inline src/_06_Intro_To_Hash_Maps/_02_LogSearch.java:76]: cool addition, I don;t think I added this case into my code ☘️

File: src/_06_Intro_To_Hash_Maps/_02_LogSearch.java [hunk 6]
```diff
@@ -1,33 +1,80 @@
...
+public class _02_LogSearch implements ActionListener {
+	HashMap<Integer, String> map = new HashMap<Integer, String>();
+	private JFrame frame = new JFrame();
+	private JPanel panel = new JPanel();
+	private JButton addButton = new JButton();
+	private JButton searchButton = new JButton();
+	private JButton viewButton = new JButton();
+	private JButton removeButton = new JButton();
+
+	public void CreateUI() {
```
Comment (mwoguri) [inline src/_06_Intro_To_Hash_Maps/_02_LogSearch.java]: How should this method name be capitalized?

File: src/_06_Intro_To_Hash_Maps/_02_LogSearch.java [hunk 7]
```diff
@@ -1,33 +1,80 @@
...
+			String userID = JOptionPane.showInputDialog(null, "Enter an ID number (3 digits)");
+			String userName = JOptionPane.showInputDialog(null, "Enter a name");
+			Integer userIDInteger = Integer.parseInt(userID);
+			map.put(userIDInteger, userName);
+		}
+		if (buttonPressed == searchButton) {
+			String userID = JOptionPane.showInputDialog(null, "Enter an ID number (3 digits)");
+			Integer userIDInteger = Integer.parseInt(userID);
+			if (map.containsKey(userIDInteger)) {
+				JOptionPane.showMessageDialog(null, "The user is " + userIDInteger);
```
Comment (mwoguri) [inline src/_06_Intro_To_Hash_Maps/_02_LogSearch.java]: [The assignment](http://central.jointheleague.org/levels/Level3/Mod1Recipes/LogSearch.html) asks you to display the user's name, not their id :)

File: src/_06_Intro_To_Hash_Maps/_02_LogSearch.java [hunk 8]
```diff
@@ -1,33 +1,80 @@
...
+			} else {
+				JOptionPane.showMessageDialog(null, "This username does not exist");
+			}
+
+		}
+		if (buttonPressed == viewButton) {
+			for (Integer i : map.keySet()) {
+				System.out.println("ID: " + i + " Name: " + map.get(i));
```
Comment (mwoguri) [inline src/_06_Intro_To_Hash_Maps/_02_LogSearch.java]: Great job on iterating over the map.  Please show the results in a dialog, not just in the console.

**Review by mwoguri — APPROVED** [review]
Great job! Please merge.

**Pull request merged**